share/revng/test/configuration/revng-qa/runtime-abi-test-toolchain.yml

```diff
@@ -0,0 +1,83 @@
+tags:
+  - name: abi-test-sources
+    implies: [has-special-compilation-requirements]
+
+  - name: abi-test-asm-sources
+    implies: [abi-test-sources]
+  - name: abi-test-c-sources
+    implies: [abi-test-sources]
+    variables:
+      CFLAGS:
+        - -std=c99
+        - -fno-pic
+        - -fno-pie
+        - -fno-stack-protector
+        - -fno-zero-initialized-in-bss
+        - -O1
+
+  - name: abi-test-argument-setup
+    implies: [abi-test-asm-sources]
+  - name: abi-test-return-value-setup
+    implies: [abi-test-c-sources]
+
+  - name: abi-test-function-prototypes
+    variables:
+      CFLAGS:
+        - -std=c99
+        - -fno-pic
+        - -fno-pie
+        - -fno-stack-protector
+        - -ggdb3
+        - -O3
+
+  - name: abi-test-symbol-gathering-script
+  - name: abi-test-list
+  - name: abi-test-printers
+  - name: abi-test-runner
+
+  - name: for-abi-runtime
+
+commands:
+  - type: revng-qa.compiled-to-object-file
+    from:
+      - type: source
+        filter: abi-test-sources
+    suffix: .o
+    command: ${TRIPLE}gcc ${INPUT} -o ${OUTPUT} ${CFLAGS} -c
+
+  # I'd rathen I didn't have to duplicate it, but it causes a dependency cycle otherwise
+  - type: revng-qa.compiled-with-debug-info-copy
+    from:
+      - type: revng-qa.compiled-to-object-file
+        filter: abi-test-return-value-setup
+      - type: revng-qa.compiled-to-object-file
+        filter: abi-test-argument-setup
+    command: ${TRIPLE}gcc ${INPUT1} ${INPUT2} -o ${OUTPUT} ${CFLAGS} -Wl,--section-start=.text=0x2000000
+
+  - type: revng-qa.disassembled-copy
+    from:
+      - type: revng-qa.compiled-with-debug-info-copy
+    suffix: .S
+    command: ${TRIPLE}objdump --wide --no-show-raw-insn $OBJDUMP_FLAGS -h -t -d $INPUT > $OUTPUT
+
+  - type: revng-qa.abi-test-extracted-addresses
+    from:
+      - type: revng-qa.disassembled-copy
+      - type: source
+        filter: abi-test-list
+      - type: source
+        filter: abi-test-symbol-gathering-script
+    suffix: .h
+    command: python3 ${INPUT3} ${INPUT1} ${INPUT2} ${OUTPUT}
```

Comment:
This command is overly complicated. I'd like to reduced it to have a single argument: `revng-qa.disassembled-copy`.
1. About `source` filtered by `abi-test-list`: the contents of these files seems to be entirely deterministic:
   1. The list of functions is fixed for all architectures and in any case we can recover them by the `objdump`.
   2. The `lfsr_seed` can be `0xDEADBEEF`.
   3. `iteration_count` is fixed.
   4. `stack_byte_count` is fixed.
   5. `generated_byte_count` can be bumped to a big enough value and become fixed. Say 2048.
2. About `source` filtered by `abi-test-symbol-gathering-script`: just hardcode the script name. I'm really confused why you decided to do this.
3. Do not create `commands` whose output is used only once. `revng-test-configure` is not a build system, we're not trying to save time on incremental builds. This is not the first time you try to use it this way, please refrain from doing this, it brings extra complexity and no benefits.
    If a command's output is used only in a single other command, squash it into it. For instance `revng-qa.disassembled-copy` can be squashed into `revng-qa.abi-test-extracted-addresses`. Review all other new commands that have been introduced as well.

Reply:
1.1. I have some FUD about that. What if a future version of gcc starts putting in something named `test_X` or whatnot? I'd rather not assume anything about the binary except for the explicit list of tests we purposefully put there.
1.2-4. Works.
1.5. I'm not happy about it, but can live with it.

2. Mostly to avoid hardcoding paths, same as `test_list`. IMO the approach should be the same for both. If you are fine with hardcoding paths here, there's no real reason to drop the list template, IMO.

3. My main reason for doing this is because this way I have an easy access to all the steps I want to see at any point. A test failed? Just pop into the directory it emits stuff into and see everything you need to see without needed to rerun it with a custom "TMP" directory or whatnot.

Reply:
1.1 Add a prefix that GCC will never use. But this is really FUD, and we're paying complexity for it. I prefer to fail in the future.
2. What's the different in hardcoding a path in a `sources:` entry or in the script? We already hardcode paths elsewhere.
   Multi input commands need to have a real reason to be used, they trigger all the auto-matching magic and they are a liability. Here we should really have a single input: the binary.
3. We have an issue with easily reproducing test by hand we need to solve, but this is not a solution. This approach leads to a lot of new commands, tags and complexity which we absolutely we need to contain. Squash everything you can. I really really don't want to split a command unless it has multiple outputs (or it takes a very long time).

Reply:
3. This makes (future) me (who's going to debug these tests going wrong) really sad, but OK.